Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift

```diff
@@ -345,7 +345,12 @@ extension WKIntelligenceReplacementTextEffectCoordinator: PlatformIntelligenceTe
 
         if chunk.finished {
             self.replacementQueue.removeFirst()
-            await self.restoreSelectionAcceptedReplacements(true)
+            if let onFlushCompletion = self.onFlushCompletion {
+                await onFlushCompletion()
+                self.onFlushCompletion = nil
+            } else {
+                await self.restoreSelectionAcceptedReplacements(true)
```

Comment:
I would add a comment above this line (above 352 and below 351) saying "At this point, the last chunk has been received, and the animation is now over, but the session has yet to end. The selection should be restored assuming it has been accepted; once the session does end, the selection will at that point then be adjusted depending on the user's action."